examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+        self.raw_train = Subset(self.dataset, self.split_idx['train'])
+        self.raw_val = Subset(self.dataset, self.split_idx['valid'])
+        self.raw_test = Subset(self.dataset, self.split_idx['test'])
+
+        self.train, self.val, self.test = [], [], []
+        for g in self.raw_train:
+            if g[0].num_nodes() > 5:
+                self.train.append(g)
+        for g in self.raw_val:
+            if g[0].num_nodes() > 5:
+                self.val.append(g)
+        for g in self.raw_test:
+            if g[0].num_nodes() > 5:
+                self.test.append(g)
+
+        print('train, test, val sizes :', len(self.train), len(self.test), len(self.val))
+        print("[I] Finished loading.")
+
+    # form a mini batch from a given list of samples = [(graph, label) pairs]
+    def collate(self, samples):
+        # The input samples is a list of pairs (graph, label).
+        graphs, labels = map(list, zip(*samples))
+        labels = torch.stack(labels)
+
+        tab_sizes_n = [ graphs[i].number_of_nodes() for i in range(len(graphs))]
+        tab_snorm_n = [ torch.FloatTensor(size,1).fill_(1./float(size)) for size in tab_sizes_n ]
+        snorm_n = torch.cat(tab_snorm_n).sqrt()
+        tab_sizes_e = [ graphs[i].number_of_edges() for i in range(len(graphs))]
+        tab_snorm_e = [ torch.FloatTensor(size,1).fill_(1./float(size)) for size in tab_sizes_e ]
+        snorm_e = torch.cat(tab_snorm_e).sqrt()
+        batched_graph = dgl.batch(graphs)
+
+        return batched_graph, labels, snorm_n, snorm_e
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--gpu_id', default=0, type=int, help="Please give a value for gpu id")
+    parser.add_argument('--seed', default=41, type=int, help="Please give a value for seed")
+    parser.add_argument('--epochs', default=450, type=int, help="Please give a value for epochs")
+    parser.add_argument('--batch_size', default=2048, type=int, help="Please give a value for batch_size")
+    parser.add_argument('--init_lr', default=0.0008, type=float, help="Please give a value for init_lr")
+    parser.add_argument('--lr_reduce_factor', default=0.8, type=float, help="Please give a value for lr_reduce_factor")
+    parser.add_argument('--lr_schedule_patience', default=8, type=int, help="Please give a value for lr_schedule_patience")
+    parser.add_argument('--min_lr', default=0.00001, type=float, help="Please give a value for min_lr")
+    parser.add_argument('--weight_decay', default=1e-5, type=float, help="Please give a value for weight_decay")
+    parser.add_argument('--L', default=4, type=int, help="Please give a value for L")
+    parser.add_argument('--hidden_dim', default=420, type=int, help="Please give a value for hidden_dim")
+    parser.add_argument('--out_dim', default=420, type=int, help="Please give a value for out_dim")
+    parser.add_argument('--dropout', default=0.2, type=float, help="Please give a value for dropout")
```

Comment:
You can drop the lines related to model configuration and simply set a default value in model class, e.g.,

```
class DGNNet(nn.Module):
    def __init__(self, avg_d, hidden_dim=420, ...):
```